app/src/test/java/com/keylesspalace/tusky/SFragmentTest.kt

```diff
@@ -0,0 +1,301 @@
+
+        fragment.onViewURL(nonMastodonQuery)
+        Assert.assertTrue(fragment.isSearching)
+        fragment.cancelActiveSearch()
+        Assert.assertFalse(fragment.isSearching)
+        emptyCallback.invokeCallback()
+        Assert.assertEquals(null, fragment.url)
+    }
+
+    @Test
+    fun search_withPreviousCancellation_completes() {
+        // begin account search
+        fragment.onViewURL(accountQuery)
+        Assert.assertTrue(fragment.isSearching)
+
+        // cancel account search
+        fragment.cancelActiveSearch()
+
+        // begin status search
+        fragment.onViewURL(statusQuery)
+        Assert.assertTrue(fragment.isSearching)
+
+        // return response from account search
+        accountCallback.invokeCallback()
+        Assert.assertEquals(null, fragment.accountId)
+
+        // ensure that status search is still ongoing
+        Assert.assertTrue(fragment.isSearching)
+        statusCallback.invokeCallback()
+        Assert.assertFalse(fragment.isSearching)
+        Assert.assertEquals(status, fragment.status)
+    }
+
+    class FakeSearchResults : Call<SearchResults>
+    {
+        var searchResults: SearchResults
+        var callback: Callback<SearchResults>? = null
+
+        constructor() {
+            searchResults = SearchResults(Collections.emptyList(), Collections.emptyList(), Collections.emptyList())
+        }
+
+        constructor(status: Status) {
+            searchResults = SearchResults(Collections.emptyList(), listOf(status), Collections.emptyList())
+        }
+
+        constructor(account: Account) {
+            searchResults = SearchResults(listOf(account), Collections.emptyList(), Collections.emptyList())
+        }
+
+        fun invokeCallback() {
+            callback?.onResponse(this, Response.success(searchResults))
+        }
+
+        override fun enqueue(callback: Callback<SearchResults>?) {
+            this.callback = callback
+        }
+
+        override fun isExecuted(): Boolean { TODO("not implemented") }
+        override fun clone(): Call<SearchResults> { TODO("not implemented") }
+        override fun isCanceled(): Boolean { TODO("not implemented") }
+        override fun cancel() { TODO("not implemented") }
+        override fun execute(): Response<SearchResults> { TODO("not implemented") }
+        override fun request(): Request { TODO("not implemented") }
+    }
+
+    class FakeSFragment : SFragment {
+        var status: Status? = null
+        var accountId: String? = null
+        var url: String? = null
+
+        constructor(): super() {
+            callList = mutableListOf()
+        }
+
+        override fun openLink(url: String) {
+            this.url = url
+        }
+
+        override fun viewAccount(id: String?) {
+            accountId = id
+        }
+
+        override fun viewThread(status: Status?) {
+            this.status = status
+        }
+
+        override fun removeItem(position: Int) { TODO("not implemented") }
```

Comment:
Android Studio will think that these are real TODOS and will show them in the TODO list. Maybe change it to `AssertionError` or something? Same for `FakeSearchResults`